src/mappings/utils/entity.ts

```diff
@@ -72,3 +72,36 @@ export function create<T extends EntityWithId>(
   Object.assign(entity, init);
   return entity;
 }
+
+export async function calculateCollectionOwnerCountAndDistribution(
+  store: Store,
+  collectionId: string,
+  newOwner?: string,
+  originalOwner?: string,
+): Promise<{ ownerCount: number; distribution: number }> {
+  const query = `
+  SELECT COUNT(DISTINCT current_owner) AS distribution,
+       COUNT(current_owner) AS owner_count
+  ${
+  newOwner
+    && `
+  ,(SELECT max(CASE
+                  WHEN current_owner = '${newOwner}' THEN 0
+                  ELSE 1
+              END)
+   FROM nft_entity) AS adjustment
+  `
+} 
+  FROM nft_entity
+  WHERE collection_id = '${collectionId}'
+  ${newOwner && `AND current_owner != '${originalOwner}'`}
+  `;
```

Comment:
This is difficult to read, I suggest this workaround (but not so pretty)
```suggestion
  let query;
  if (newOwner && originalOwner) {
    query = `SELECT 
      COUNT(DISTINCT current_owner) AS distribution,
      COUNT(current_owner) AS owner_count,
      (
        SELECT max(
          CASE
            WHEN current_owner = '${newOwner}' THEN 0
            ELSE 1
          END
        )
        FROM nft_entity
      ) AS adjustment
      FROM nft_entity
      WHERE collection_id = '${collectionId}'
      AND current_owner != '${originalOwner}'`;
  }

  query = `SELECT 
      COUNT(DISTINCT current_owner) AS distribution,
      COUNT(current_owner) AS owner_count
      FROM nft_entity
      WHERE collection_id = '${collectionId}'`;
```

Reply:
pfff leeeet me fix that

Reply:
new event version?

Reply:
Fixed in  #173

Reply:
@roiLeo did you maybe forgot to rebuild? I've just run `make process` after building and didn't encounter the error. I applied your suggestion.